pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go

```diff
@@ -418,11 +420,11 @@ func (a *index) serviceEntryWorkloadBuilder(
 			var waypointAddress *workloadapi.GatewayAddress
 			// Endpoint does not have a real ObjectMeta, so make one
 			if !implicitEndpoints {
-				if waypoint := fetchWaypointForWorkload(ctx, waypoints, namespaces, metav1.ObjectMeta{
+				if waypoint, err := fetchWaypointForWorkload(ctx, waypoints, namespaces, metav1.ObjectMeta{
 					Name:      se.Name,
 					Namespace: se.Namespace,
 					Labels:    wle.Labels,
-				}); waypoint != nil {
+				}); err != nil {
```

Comment:
why this change? Why not updating `waypointAddress` in non error case?

Reply:
:+1: fixed in https://github.com/istio/istio/pull/53038

Reply:
Thanks 👍